youtube_dl/postprocessor/ffmpeg.py

```diff
@@ -215,6 +215,66 @@ def _ffmpeg_filename_argument(self, fn):
         return 'file:' + fn if fn != '-' else fn
 
 
+class FFmpegCutVideoPP(FFmpegPostProcessor):
+    def __init__(self, downloader=None, startTime=0, endTime=None):
+        FFmpegPostProcessor.__init__(self, downloader)
+        self._startTime = startTime
+        self._endTime = endTime
+
+    def toTime(self, seconds):
+        m, s = divmod(seconds, 60)
+        h, m = divmod(m, 60)
+        return "%d:%02d:%02d" % (h, m, s)
+
+    def run(self, information):
+        if self._startTime == 0 and self._endTime == None:
+            self._downloader.to_screen('[ffmpeg] No start time or end time. Keeping original')
+            return [], information
+
+        if self._endTime and self._endTime <= self._startTime:
+            raise PostProcessingError("end time smaller or equal to the start time")
+
+        if self._endTime == 0:
+            raise PostProcessingError("end time can't be zero")
+
+        duration = information.get('duration')
+
+        if self._endTime and duration and self._endTime >= duration:
+            self._downloader.to_screen('WARNING: end time greater than video duration')
+            self._endTime = None
+
+        options = ['-c', 'copy']
+        message = '[ffmpeg] Cutting video '
+
+        if self._startTime:
+            start = self.toTime(self._startTime)
+            options.extend(['-ss', start])
+            message += 'from %s ' % (start)
+            if duration:
+                duration -= self._startTime
+
+        if self._endTime:
+            end = self.toTime(self._endTime)
+            options.extend(['-to', end])
+            message += 'to %s' % (end)
+            if duration:
+                duration -= self._endTime
+
+        if '-to' not in options and '-ss' not in options:
+            self._downloader.to_screen('[ffmpeg] Nothing to cut. Keeping original')
+            return [], information
+
+        path = information['filepath']
+        temp_filename = prepend_extension(path, 'temp')
+        self._downloader.to_screen(message)
+        if duration:
+            information['duration'] = duration
```

Comment:
This has no effect. Postprocessors work on info dict copy.

Reply:
I've seen this used [here](https://github.com/marnunez/youtube-dl/blob/afc763ee2e193b5711e9af53b1e133031611e72d/youtube_dl/postprocessor/ffmpeg.py#L299) and [here](https://github.com/marnunez/youtube-dl/blob/afc763ee2e193b5711e9af53b1e133031611e72d/youtube_dl/postprocessor/ffmpeg.py#L354) with filepath and other keys. Does this work in this cases?